hooks/tk-multi-loader2/tk-nuke_actions.py

```diff
@@ -100,9 +130,128 @@ def execute_action(self, name, params, sg_publish_data):
         if name == "deep_read_node":
             self._create_deep_read_node(path, sg_publish_data)
 
+        if name == "clip_img_mov":
+            self._import_img_mov(path, sg_publish_data)
+
+        if name == "clip_img_seq":
+            self._import_img_seq(path, sg_publish_data)
+
+        # the code is same as adding an image sequence
+        if name == "clip_import_bin":
+            self._import_img_seq(path, sg_publish_data)
+
     ##############################################################################################################
     # helper methods which can be subclassed in custom hooks to fine tune the behavior of things
 
+    def _import_img_mov(self, path, sg_publish_data):
+        """
+        Imports the given publish data into Nuke Studio or Hiero as a clip.
+
+        :param str path: Path to the file(s) to import.
+        :param dict sg_publish_data: Shotgun data dictionary with all of the standard publish
+            fields.
+        """
+        if not self.parent.engine.studio_enabled and not self.parent.engine.hiero_enabled:
+            raise Exception("Importing shot clips is only supported in Hiero and Nuke Studio.")
+
+        if not hiero.core.projects():
+            raise Exception("An active project must exist to import clips into.")
+
+        # read published data
+        image_path = sg_publish_data['path']['local_path']
+        version_name = sg_publish_data['version']['name']
+
+        # get pipeline step (Shot context)
+        # eg. 'RND_000100_layout_bg.v009'
+        department = version_name.split('_')[2]
```

Comment:
Can we not get this from the `sg_publish_data` dict? Like from the associated Task? We should not be using string parsing to get this information. If its not already available in `sg_publish_data`, you should do a Shotgun query to get the field value.